src/hooks/transactions/batch/useTrackBatchTransactions.ts

```diff
@@ -0,0 +1,153 @@
+};
+
+export const useTrackBatchTransactions = ({
+  apiAddress,
+  batchId,
+  onSuccess,
+  onFail
+}: TrackBatchTransactionsStatusProps) => {
+  const dispatch = useDispatch();
+  const stopPollingRef = useRef<boolean>(true);
+
+  const updateBatchTransactionsStatuses = useUpdateBatchTransactionsStatuses();
+  const { batches } = useGetBatchesTransactions();
+  const { address } = useGetAccount();
+
+  const batchTransactions = useMemo(() => {
+    if (!batchId) {
+      return {};
+    }
+
+    return batches[batchId] ?? {};
+  }, [batchId, batches]);
+
+  const batchStatus = useMemo(() => {
+    if (!batchId) {
+      return '';
+    }
+
+    return batches[batchId]?.status ?? '';
+  }, [batchId, batches]);
+
+  const getBatchStatus = useCallback(
+    async (id: string) => {
+      try {
+        return await getBatchTransactionsStatus({
+          apiAddress,
+          batchId: id,
+          address
+        });
+      } catch (error) {
+        console.error(error);
+        return null;
+      }
+    },
+    [apiAddress, address]
+  );
+
+  const verifyBatchStatus = useCallback(
+    async ({ batchId }: { batchId: string }) => {
+      const batchStatus = await getBatchStatus(batchId);
+
+      if (!batchStatus) {
+        stopPollingRef.current = true;
+        return;
+      }
+
+      dispatch(updateBatchTransactions(batchStatus));
+
+      const isBatchSuccessful =
+        batchStatus?.status === BatchTransactionStatus.success;
+      const isBatchFailed =
+        batchStatus?.status === BatchTransactionStatus.invalid ||
+        batchStatus?.status === BatchTransactionStatus.dropped;
+      const isBatchNotFound =
+        Boolean(batchStatus?.statusCode) && Boolean(batchStatus?.message);
+
+      if (isBatchSuccessful) {
+        stopPollingRef.current = true;
+        onSuccess?.(batchId);
+        return;
+      }
+
+      if (isBatchFailed) {
+        stopPollingRef.current = true;
+        onFail?.(
+          batchId,
+          `Error processing batch transactions. Status: ${batchStatus?.status}`
+        );
+        return;
+      }
+
+      if (isBatchNotFound) {
+        stopPollingRef.current = true;
+      }
+    },
+    [getBatchStatus, onSuccess, onFail, stopPollingRef.current]
+  );
+
+  const onMessage = useCallback(() => {
+    // Do nothing, used for backwards compatibility to avoid breaking changes
+    // TODO: Will be removed in the next major release
+  }, []);
+
+  const onBatchUpdate = useCallback(
+    async (data: BatchTransactionsWSResponseType) => {
+      await verifyBatchStatus({ batchId: data.batchId });
+
+      await updateBatchTransactionsStatuses({
+        batchId: data.batchId,
+        shouldRefreshBalance: true
+      });
+    },
+    [verifyBatchStatus, updateBatchTransactionsStatuses]
+  );
+
+  useRegisterWebsocketListener(onMessage, onBatchUpdate);
+
+  useEffect(() => {
+    const interval = setTimeout(async () => {
+      stopPollingRef.current = false;
+    }, 30000);
```

Comment:
I think we have some global constants for these intervals. If not, maybe there are needed